lib/rubocop/cop/rails/read_write_attribute.rb

```diff
@@ -66,10 +66,13 @@ def on_send(node)
         private
 
         def within_shadowing_method?(node)
-          node.each_ancestor(:def).any? do |enclosing_method|
-            shadowing_method_name = node.first_argument.value.to_s
-            shadowing_method_name << '=' if node.method?(:write_attribute)
+          first_arg = node.first_argument
+          return false unless first_arg.respond_to?(:value)
+
+          shadowing_method_name = first_arg.value.to_s
+          shadowing_method_name << '=' if node.method?(:write_attribute)
 
+          node.each_ancestor(:def).any? do |enclosing_method|
```

Comment:
Unrelated to this PR.
Wondering, why we need to check all ancestors, or is it sufficient to check if the first one:
 - exists
 - matches the name

Reply:
Actually you are right, we do indeed interested in the first `def` ancestor
I guess `any?` is an implicit way of checking both exists and matches the way
Let me try to make it more explicit!

Reply:
Really good catch, I've changed it! Thanks ❤️